README.md

```diff
@@ -52,20 +54,143 @@ Install by adding `rbac` to your list of dependencies in `mix.exs`:
 ```elixir
 def deps do
   [
-    {:rbac, "~> 0.1.0"}
+    {:rbac, "~> 0.3.0"}
   ]
 end
 ```
 
-API/Function reference available at
-[https://hexdocs.pm/rbac](https://hexdocs.pm/rbac).
+### Setup
+
+Open the `application.ex` file of your project 
+and locate the `def start(_type, _args) do` definition, e.g:
+
+```elixir
+def start(_type, _args) do
+  # List all child processes to be supervised
+  children = [
+    # Start the Ecto repository
+    Auth.Repo,
+    # Start the endpoint when the application starts
+    {Phoenix.PubSub, name: Auth.PubSub},
+    AuthWeb.Endpoint
+    # Starts a worker by calling: Auth.Worker.start_link(arg)
+    # {Auth.Worker, arg},
+  ]
+
+  # See https://hexdocs.pm/elixir/Supervisor.html
+  # for other strategies and supported options
+  opts = [strategy: :one_for_one, name: Auth.Supervisor]
+  Supervisor.start_link(children, opts)
+end
+```
 
+Add the following code at the top of the function definition:
+
+```elixir
+# initialize RBAC Cache:
+RBAC.init_roles_cache(
+  "https://dwylauth.herokuapp.com",
+  AuthPlug.Token.client_id()
```

Comment:
This makes `rbac` implicitly dependent on `auth_plug` - do we want to do this?

Reply:
Yeah, I thought about this and yes, I don't see _us_ using the modules independently. 💭 
Ultimately both `auth_plug` and `rbac` coupled with `auth` for better or worse.
I just like having separate modules/repos for maintenance purposes. (_they can be independently extended/maintained_)
If someone using any of our modules asks to use them independently, we can always refactor. 

I briefly thought of putting the `init_roles_cache` function in the `auth_plug` ... would that be better? 🤷‍♂️ 

You _have_ reminded me that I haven't explicitly mentioned getting an `AUTH_API_KEY`  in the `rbac/README.md` ... 🤦

Reply:
Putting `init_roles_cache` in auth_plug may be better as then all the functions that work directly with the `auth` server are contained in one place.

However, this also works as it means you can just ignore `rbac` if you want a simple authentication solution, which works for ~80% of people.

Having though about this some more, from a DevX point of view, as well as a "seperation of concerns" point of view, I think its okay to have the dependency on `auth_plug` - but it should be made clear that this *extends* `auth plug`.

Reply:
@th0mas technically the person doesn't need to depend on `auth_plug` at all.
They can just do the following:

```
apikey = System.get_env("AUTH_API_KEY")
RBAC.init_roles_cache(
  "https://dwylauth.herokuapp.com",
  List.first(String.split(apikey, "/"))
)
```

I just _assume_ that they are using `rbac` with `auth_plug` because that's _our_ current use-case. 💭

Reply:
also worth noting that in `Elixir 1.11` (_shipping soon_: https://www.youtube.com/watch?v=guc16D_0Imk ...) this type of dependency will need to be explicit. We can no longer _assume_ peer dependencies. So maybe we should update the docs/instructions to this exact code. That way someone can use RBAC 100% independently of `auth_plug`. 👍

Reply:
I've created a function `insert_roles_into_ets/1` that can be called with a list of roles and added docs/instructions to the `README.md` for how to use it. This makes `RBAC` 100% independent from both `auth` and `auth_plug`. 
It also means that anyone using `auth_plug` doesn't need to use `rbac`. Everything is independent but complimentary. 👍

Reply:
I've created a function and updated the docs so that `auth_plug` is no longer assumed/required
and people can use `rbac` 100% independently of `auth`:
```elixir
roles = [%{id: 1, name: "admin"}, %{id: 2, name: "subscriber"}]
RBAC.insert_roles_into_ets_cache(roles)
```